app/src/main/kotlin/com/wire/android/media/audiomessage/ConversationAudioMessagePlayer.kt

```diff
@@ -34,14 +36,46 @@ import kotlinx.coroutines.flow.filterNotNull
 import kotlinx.coroutines.flow.flow
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.merge
+import kotlinx.coroutines.flow.onStart
 import kotlinx.coroutines.launch
 import javax.inject.Inject
+import javax.inject.Singleton
 
-class ConversationAudioMessagePlayer
+@Singleton
+class ConversationAudioMessagePlayerProvider
 @Inject constructor(
     private val context: Context,
     private val audioMediaPlayer: MediaPlayer,
-    private val getMessageAsset: GetMessageAssetUseCase
+    @KaliumCoreLogic private val coreLogic: CoreLogic,
+) {
+    private var player: ConversationAudioMessagePlayer? = null
+    private var usageCount: Int = 0
+
+    @Synchronized
+    fun provide(): ConversationAudioMessagePlayer {
+        if (player == null) {
+            player = ConversationAudioMessagePlayer(context, audioMediaPlayer, coreLogic)
+        }
+        usageCount++
+
+        return player!!
```

Comment:
[Just a thought] Im not a big fan of the `!!` operator. so maybe something like:
```
val player = player ?: ConversationAudioMessagePlayer(context, audioMediaPlayer, coreLogic).also { player = it }
usageCount++

return player
```